sync_check/util/chunk.go

```diff
@@ -0,0 +1,313 @@
+			chunks = append(chunks, r)
+			cutoff += step
+		}
+		log.Debugf("getChunksForTable cut table: cnt=%d min=%v max=%v step=%v chunk=%d",
+			count, min, max, step, len(chunks))
+	} else {
+		max, ok1 := chunk.end.(string)
+		min, ok2 := chunk.begin.(string)
+		if !ok1 || !ok2 {
+			return nil, errors.Errorf("can't parse chunk's begin: %v, end: %v", chunk.begin, chunk.end)
+		}
+
+		// get random value as split value
+		splitValues, err := GetRandomValues(db, dbname, table, column.Name, count-1, min, max, timeRange)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		var minTmp, maxTmp string
+		var i int64
+		for i = 0; i < count; i++ {
+			if i == 0 {
+				minTmp = min
+				maxTmp = splitValues[i]
+			} else if i == int64(len(splitValues)) {
+				minTmp = splitValues[i-1]
+				maxTmp = max
+			} else {
+				minTmp = splitValues[i-1]
+				maxTmp = splitValues[i]
+			}
+			r := newChunkRange(minTmp, maxTmp, true, false, true)
+			chunks = append(chunks, r)
+		}
+		log.Debugf("[dumper] getChunksForTable cut table: cnt=%d min=%s max=%s chunk=%d",
+			count, min, max, len(chunks))
+	}
+	chunks[0].containB = chunk.containB
+	chunks[len(chunks)-1].containE = chunk.containE
+	return chunks, nil
+}
+
+func findSuitableField(db *sql.DB, dbname string, table string) (*schema.TableColumn, error) {
+	// first select the index, and number type index first
+	column, err := FindSuitableIndex(db, dbname, table, true)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if column != nil {
+		return column, nil
+	}
+	log.Infof("%s.%s don't have index, will use a number column as split field", dbname, table)
+
+	// select a number column
+	column, err = FindNumberColumn(db, dbname, table)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if column != nil {
+		return column, nil
+	}
+	log.Infof("%s.%s don't have number columns, will use the first column as split field", dbname, table)
+
+	// use the first column
+	column, err = GetFirstColumn(db, dbname, table)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if column != nil {
+		return column, nil
+	}
+
+	return nil, errors.Errorf("no column find in table %s.%s", dbname, table)
+}
+
+func generateDumpJob(db *sql.DB, dbname, tableName, timeField, beginTime, endTime, splitField string, chunkSize int, sample int) ([]*dumpJob, error) {
+	jobBucket := make([]*dumpJob, 0, 10)
+	var jobCnt int
+	var column *schema.TableColumn
+	var err error
+
+	if splitField == "" {
+		// find a column for split data
+		column, err = findSuitableField(db, dbname, tableName)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	} else {
+		var table *schema.Table
+		table, err = GetSchemaTable(db, dbname, tableName)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		exist := false
+		column, exist = GetColumnByName(table, splitField)
+		if !exist {
+			return nil, fmt.Errorf("can't find column %s in table %s", splitField, tableName)
+		}
+	}
+
+	timeRange := "true"
```

Comment:
set timeRange to "true" can make the code more simple, no need to judge the timeRange's value.

Reply:
It's better to comment it in the code.